source/opt/ir_context.h

```diff
@@ -795,7 +795,7 @@ void IRContext::AddAnnotationInst(std::unique_ptr<Instruction>&& a) {
 void IRContext::AddType(std::unique_ptr<Instruction>&& t) {
   module()->AddType(std::move(t));
   if (AreAnalysesValid(kAnalysisDefUse)) {
-    get_def_use_mgr()->AnalyzeInstDef(&*(--types_values_end()));
+    get_def_use_mgr()->AnalyzeInstDefUse(&*(--types_values_end()));
```

Comment:
Why are we analysing def-use now?  I don't have an issue with it, but I'm not sure I understand why we need this now.

Reply:
When you analyze just the definitions, it clears all of the uses.  Then the uses in the type definition are not present.  This lead to the def-use chains being inconsistent.  I think that `AddType` should do nothing or do both.  Only doing the definition is odd.